### PR TITLE
Staging and Prod hosts are parameterized in accesstype.js

### PR DESCRIPTION
**Pull request by cricket-lover**
# Description

Parameterized the staging and prod hosts in accesstype.js. Now the hosts are configurable. 
Refererence to the issue-> https://github.com/quintype/ace-planning/issues/345

## Type of change

- [ ] This change requires a documentation update

### CODE REVIEW

File: src/components/access-type.js [hunk 1]
```diff
@@ -9,28 +9,31 @@ import {
...
-const prod_Host = 'https://www.accesstype.com';
-const staging_Host = 'https://staging.accesstype.com';
+} from "../store/actions";
+import PropTypes from "prop-types";
+import { awaitHelper } from "../utils";
 
 class AccessTypeBase extends React.Component {
+  constructor(props) {
+    super(props);
+    this.prod_Host = props.prodHost || "https://www.accesstype.com";
```
Comment (nandap4790) [inline src/components/access-type.js:19]: can you change the casing ?

Comment (cricket-lover) [reply src/components/access-type.js:19]: This is how it has been used the all over the places before so i kept the same.
But sure we can make it camel case

File: src/components/access-type.js [hunk 2]
```diff
@@ -551,6 +557,8 @@ const mapDispatchToProps = dispatch => ({
  *                  email={email}
  *                  phone={phone}
  *                  disableMetering={disableMetering}
+ *                  prodHost="www.abc.com"
+ *                  stagingHost="staging.abc.com"
```
Comment (ags1773) [inline src/components/access-type.js:561]: You need to give a short explanation of what's `prodHost` and `stagingHost`.. like how it's given on line 514 to 519

File: src/components/access-type.js [hunk 3]
```diff
@@ -542,6 +548,10 @@ const mapDispatchToProps = dispatch => ({
  * ```javascript
  * import { AccessType } from "@quintype/components";
  *
+ * propName | Description
+ *  --- | ---
+ *  prodHost | It is passed to configure the production host.(Optional)
+ *  stagingHost | It is passed to configure the staging host.(Optional)
```
Comment (nandap4790) [inline src/components/access-type.js:554]: The format does not seem to align with the existing docs right ?
@ags1773

File: src/components/access-type.js [hunk 4]
```diff
@@ -476,24 +479,27 @@ AccessTypeBase.propTypes = {
   enableAccesstype: PropTypes.bool.isRequired,
   accessTypeKey: PropTypes.string.isRequired,
   accessTypeBkIntegrationId: PropTypes.string.isRequired,
+  prodHost: PropTypes.string,
+  stagingHost: PropTypes.string,
```
Comment (Deokumarjnu) [inline src/components/access-type.js:483]: is it possible to take only one host and do the logic based on it ?
something like
```
isStaging: true,
host: staging.abc.com
```
or 
```
isStaging: false
host: abc.com
```

File: src/components/access-type.js (hunk 3 above)
Comment (ags1773) [reply src/components/access-type.js:554]: @cricket-lover pls move this block of code on top, right after `AccessType is a generic connected render prop which exposes methods to handle access to stories / assets and initialize accesstype js`

File: src/components/access-type.js (hunk 2 above)
Comment (cricket-lover) [reply src/components/access-type.js]: @nandap4790 Default value as in?

File: src/components/access-type.js (hunk 4 above)
Comment (cricket-lover) [reply src/components/access-type.js:483]: Yeah we can do that, but then the props cannot be optional right

File: src/components/access-type.js (hunk 2 above)
Comment (nandap4790) [reply src/components/access-type.js]: what is the fallback value if there is no value passed from the frontend

Comment (ags1773) [reply src/components/access-type.js]: @cricket-lover 
```
 *  propName    |   Description
 *  --------    |   -----------
 *  prodHost    |   It is passed to configure the production host.(Optional, default value is "https://www.accesstype.com")
 *  stagingHost |   It is passed to configure the staging host.(Optional, default value is "https://staging.accesstype.com")
 *
```

**Review by nandap4790 — APPROVED** [review]

**Review by ags1773 — APPROVED** [review]

**Review by Deokumarjnu — APPROVED** [review]

**Pull request merged**